### PR TITLE
Fix missing casts in Enumerable.Select assert

### PR DESCRIPTION
**Pull request by timandy**
Fix #37410 missed casts.

### CODE REVIEW

File: src/System.Linq/src/System/Linq/Select.SpeedOpt.cs [hunk 1]
```diff
@@ -171,7 +171,7 @@ private sealed partial class SelectRangeIterator<TResult> : Iterator<TResult>, I
             public SelectRangeIterator(int start, int end, Func<int, TResult> selector)
             {
                 Debug.Assert(start < end);
-                Debug.Assert((end - start) <= int.MaxValue);
+                Debug.Assert(unchecked((uint)(end - start) <= (uint)int.MaxValue));
```
Comment (stephentoub) [inline src/System.Linq/src/System/Linq/Select.SpeedOpt.cs]: The `unchecked` isn't needed. `unchecked` is the default.

Comment (timandy) [reply src/System.Linq/src/System/Linq/Select.SpeedOpt.cs]: Just make easy to read. Many other places used `unchecked` too.

Comment (stephentoub) [inline src/System.Linq/src/System/Linq/Select.SpeedOpt.cs]:
```suggestion
                Debug.Assert((uint)(end - start) <= (uint)int.MaxValue);
```

Comment (timandy) [reply src/System.Linq/src/System/Linq/Select.SpeedOpt.cs]: OK, I will force push to change the commit.

Comment (timandy) [reply src/System.Linq/src/System/Linq/Select.SpeedOpt.cs]: Done.

**Comment by stephentoub** [comment]
Thanks!

**Pull request merged**